Source/SuperLinq/Join.LoopJoin.cs

```diff
@@ -0,0 +1,69 @@
+﻿namespace SuperLinq;
+
+public static partial class SuperEnumerable
+{
+	private static IEnumerable<TResult> LoopJoin<TLeft, TRight, TKey, TResult>(
+		IEnumerable<TLeft> left,
+		IEnumerable<TRight> right,
+		JoinOperation joinOperation,
+		Func<TLeft, TKey> leftKeySelector,
+		Func<TRight, TKey> rightKeySelector,
+		Func<TLeft, TResult>? leftResultSelector,
+		Func<TLeft, TRight, TResult> bothResultSelector)
+	{
+		foreach (var l in left)
+		{
+			var lKey = leftKeySelector(l);
+			var flag = false;
+
+			foreach (var r in right)
+			{
+				var rKey = rightKeySelector(r);
+				if (EqualityComparer<TKey>.Default.Equals(lKey, rKey))
+				{
+					flag = true;
+					yield return bothResultSelector(l, r);
+				}
+			}
+
+			if (joinOperation.HasFlag(JoinOperation.LeftOuter)
+				&& !flag)
+			{
```

Comment:
Instead of duplicate code for the comparer-less overloads, why not just forward the call to an overload with a `EqualityComparer.Default`? (Q here and in the other places)

Reply:
I don't think it's worthwhile for this library. LINQ is already considered not a solution for high-performance scenarios, and the BLC in general needs to hold itself to a higher standard. I'd focus for now on maintainability.